### DISCUSSION TITLE
Does the example of creating the GraphConv class by extending the MessagePassing base class correspond to the actual implementation of the GraphConv layer in the library?

### DISCUSSION THREAD

**Discussion by giuseppefutia** [pyg-team/pytorch_geometric#9217] [General]
Hello,

For using the GCNConv layer, I could extend the MessagePassing base class as shown here:
https://pytorch-geometric.readthedocs.io/en/latest/notes/create_gnn.html.

Or, I could use the actual implementation: https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.models.GCN.html.

Beyond the differences in terms of efficiency, are these implementations the same? So, should I expect the same result considering the same input features and weights?

I think that for training and learning purposes it could be helpful to show that the two implementations lead to the same result.

Thank you!

**Comment by rusty1s** [comment]
Yes, the two implementations are mostly equivalent. Modules in PyG are usually a bit advanced than basic than the models given in tutorials (e.g., they support different sparse tensor input formats, support TorchScript, etc).

**Reply by giuseppefutia** [reply]
Thanks for the clarification, @rusty1s!
